ckb/src/main/java/org/nervos/ckb/transaction/handler/OmnilockScriptHandler.java

```diff
@@ -0,0 +1,155 @@
+package org.nervos.ckb.transaction.handler;
+
+import org.nervos.ckb.Network;
+import org.nervos.ckb.sign.ScriptGroup;
+import org.nervos.ckb.sign.omnilock.OmnilockIdentity;
+import org.nervos.ckb.sign.omnilock.OmnilockWitnessLock;
+import org.nervos.ckb.sign.signer.OmnilockSigner;
+import org.nervos.ckb.sign.signer.Secp256k1Blake160MultisigAllSigner;
+import org.nervos.ckb.transaction.AbstractTransactionBuilder;
+import org.nervos.ckb.type.CellDep;
+import org.nervos.ckb.type.Script;
+import org.nervos.ckb.type.WitnessArgs;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+public class OmnilockScriptHandler implements ScriptHandler {
+  private CellDep cellDep;
+  private CellDep singleSignCellDep;
+  private CellDep multiSignCellDep;
+  private byte[] codeHash;
+
+  public OmnilockScriptHandler() {
+  }
+
+  public OmnilockScriptHandler(Network network) {
+    if (network == Network.MAINNET) {
+      codeHash = Script.OMNILOCK_CODE_HASH_MAINNET;
+      cellDep = new CellDep("0xdfdb40f5d229536915f2d5403c66047e162e25dedd70a79ef5164356e1facdc8", 0, CellDep.DepType.CODE);
+      singleSignCellDep = new CellDep("0x71a7ba8fc96349fea0ed3a5c47992e3b4084b031a42264a018e0072e8172e46c", 0, CellDep.DepType.DEP_GROUP);
+      multiSignCellDep = new CellDep("0x71a7ba8fc96349fea0ed3a5c47992e3b4084b031a42264a018e0072e8172e46c", 1, CellDep.DepType.DEP_GROUP);
+    } else if (network == Network.TESTNET) {
+      codeHash = Script.OMNILOCK_CODE_HASH_TESTNET;
+      cellDep = new CellDep("0x27b62d8be8ed80b9f56ee0fe41355becdb6f6a40aeba82d3900434f43b1c8b60", 0, CellDep.DepType.CODE);
+      singleSignCellDep = new CellDep("0xf8de3bb47d055cdf460d93a2a6e1b05f7432f9777c8c474abf4eec1d4aee5d37", 0, CellDep.DepType.DEP_GROUP);
+      multiSignCellDep = new CellDep("0xf8de3bb47d055cdf460d93a2a6e1b05f7432f9777c8c474abf4eec1d4aee5d37", 1, CellDep.DepType.DEP_GROUP);
+    } else {
+      throw new IllegalArgumentException("Unsupported network: " + network);
```

Comment:
Init an instance by constructor `OmnilockScriptHandler()` and use methods like `setCellDep()`, `setCodeHash()`, `setSingleSignCellDep()` and so on.

Reply:
Random thoughts: at the code level we can separate a script implementation into these parts
- `script info`: information that all scripts have. they are cellDep and codeHash.
- `script configuration`: an abstract representation decoding from script args, like config in omnilock args.
- `signing configuration`: a representation to build the correct witness placeholder or help to sign. These data should be provided by the user. Like multi-sign config.
- `signing secret`: the secret to signing transactions, like a secret key or SMT proof. It may need the helps of `script configuration` (decoding from args) or `signing configuration` (provided by the user) to sign.

The above is logical separation and some can be simplified. For example, the `signing configuration` for single-sign is that `signature is 65-byte`, so we can hard-code it. But for multi-sign cases, it should be provided by the user.

Reply:
Another thought is that script like omnilock has multiple logical branches. It may be a good way to provide each branch with a signer (just like regarding each branch as a script in code), and then implement an aggregated signer to provide logic to simplify work like choosing which branch to sign like the current implementation when needed.

I think as CKB VM is very flexible it is inevitable to have a complex script like omnilock and in the future there may be more and more. From the perspective of developer it is a disaster to maintain and in the user level it will be a black box and lose fine-grained control.  Break up code is the way.